src/sentry/workflow_engine/endpoints/organization_detector_details.py

```diff
@@ -210,6 +214,9 @@ def delete(self, request: Request, organization: Organization, detector: Detecto
         if detector.type == MetricIssue.slug:
             schedule_update_project_config(detector)
 
+        RegionScheduledDeletion.schedule(detector, days=0, actor=request.user)
+        detector.update(status=ObjectStatus.PENDING_DELETION)
```

Comment:
it looks like this is already being called [here](https://github.com/getsentry/sentry/pull/101843/files#diff-d1afe0faea0f9882d9a5a0dcea0a3f64c6059f8561cd8611e4bc8d12a8e17b96R209) -- does the the life cycle hook need to be called before the pending deletion happens? (the hook is currently invoked after the detector is updated to be pending deletion)